card/mod/admin/set/self/admin.rb

```diff
@@ -11,6 +15,10 @@
   when :repair_permissions   then Card.repair_all_permissions
   when :clear_solid_cache    then Card.clear_solid_cache
   when :clear_machine_cache  then Card.reset_all_machines
+  else
+    if (task_data = Hash[tasks][task])
+      task_data[:execute_policy].call
+    end
```

Comment:
I like this.  wondering if we should register all the above tasks this way, too (at some point)